### PR TITLE
refactor(dop): add unit-test pb provider

### PR DESCRIPTION
**Pull request by chengjoey**
#### What this PR does / why we need it:
add unit-test pb

#### Which issue(s) this PR fixes:

- Fixes #your-issue_number
- [Erda Cloud Issue Link](https://erda.cloud/erda/dop/projects/387/issues/gantt?filter__urlQuery=eyJhc3NpZ25lZSI6WyIxMDAxMjA1Il19&id=309424&iterationID=1190&pId=306322&type=TASK)

#### Specified Reviewers:

/assign @Effet 

#### ChangeLog

| Language | Changelog |
| --------- | ------------ |
| 🇺🇸 English |  add unit-test pb            |
| 🇨🇳 中文    |    添加单元测试相关的pb文件          |

#### Need cherry-pick to release versions?

Add comment like `/cherry-pick release/1.0` when this PR is merged.

> For details on the cherry pick process, see the [cherry pick requests](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md#how-to-cherry-pick-a-merged-pr) section under [CONTRIBUTING.md](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by chengjoey** [comment]
tested in dev
image

File: api/proto/dop/qa/unittest/unittest.proto [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+    option (erda.common.openapi) = {
+      path : "/api/qa/test/{id}",
+    };
+  }
+}
+
+message TestRecordPagingRequest {
+  int64 pageNo = 1;
+  int64 pageSize = 2;
+  uint64  applicationId =3;
```
Comment (Effet) [inline api/proto/dop/qa/unittest/unittest.proto:54]: bad format

File: apistructs/qa.go [hunk 2]
```diff
@@ -186,7 +186,7 @@ type Test struct {
 	Classname string        `json:"classname" yaml:"classname"`
 	Duration  time.Duration `json:"duration" yaml:"duration"`
 	Status    TestStatus    `json:"status" yaml:"status"`
```
Comment (Effet) [inline apistructs/qa.go:188]: is this apistructs useless? shall we delete it?

**Review by Effet — CHANGES_REQUESTED** [review]

File: api/proto/dop/qa/unittest/unittest.proto (hunk 1 above)
Comment (chengjoey) [reply api/proto/dop/qa/unittest/unittest.proto:54]: i had `make format`, there are what it looks like after formatting
image
other protos are the same format

File: apistructs/qa.go (hunk 2 above)
Comment (chengjoey) [reply apistructs/qa.go:188]: apistructs are used by ut-action

**Comment by chengjoey** [comment]
image

File: apistructs/qa.go (hunk 2 above)
Comment (chengjoey) [reply apistructs/qa.go]: image
pipeline and some modules are uesd

Comment (chengjoey) [reply apistructs/qa.go]: adjuste ut-action use pb, and some pkg use pb

File: api/proto/dop/qa/unittest/unittest.proto (hunk 1 above)
Comment (chengjoey) [reply api/proto/dop/qa/unittest/unittest.proto:54]: adjusted, make format by protobuf plugin

**Review by Effet — CHANGES_REQUESTED** [review]

**Review by Effet — DISMISSED** [review]

File: .erda/migrations/qa/20220519-qa-test-record-report.sql [hunk 3]
```diff
@@ -0,0 +1 @@
+ALTER TABLE qa_test_records ADD COLUMN `report` longtext COMMENT '覆盖率报告';
```
[the remark this answers is not in the document]
Comment (chengjoey) [reply .erda/migrations/qa/20220519-qa-test-record-report.sql]: adjusted

**Review by Effet — APPROVED** [review]

**Pull request merged**